python/triton/language/core.py

```diff
@@ -1265,7 +1265,7 @@ def __str__(self) -> str:
         return f"tensor_descriptor<{self.type}>"
 
     @builtin
-    def load(self, offsets: List[tensor], _builder=None) -> tensor:
+    def load(self, offsets: List[int], _builder=None) -> tensor:
```

Comment:
Technically this should be `constexpr | tensor`, but certainly not `int`.

Reply:
I guess there is a question of if we could make our type annotations work for both the kernel dev and for the frontend dev. `int` would probably work better in mypy when writing a kernel, even though it's wrong.

Reply:
`constexpr` yes, but is `tensor` correct? The underlying op expects a `ArrayRef<Value>`

Reply:
Oh I see, a scalar tensor

Reply:
I'll change it to `constexpr | tensor` instead to avoid changing the convention so far.

Reply:
I32 is a runtime value, which in the frontend will be represented as a `tensor` with a scalar type.

Reply:
Gotcha. Thanks!